uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -688,6 +688,111 @@ private function getEntitlementCalculationMock(
+			date('YmdHis')
+		);
+
+		$this->assertEquals(4, $periodEntitlement->getFutureBalance());
+	}
+
+	public function testGetPeriodEntitlementsForContact() {
+    LeavePeriodEntitlement::create([
+      'period_id' => 1,
+      'type_id' => 1,
+      'contact_id' => 1
+    ]);
+
+    LeavePeriodEntitlement::create([
+      'period_id' => 1,
+      'type_id' => 2,
+      'contact_id' => 1
+    ]);
+
+    $entitlements = LeavePeriodEntitlement::getPeriodEntitlementsForContact($contact_id = 1, $period_id = 1);
+    $this->assertEquals(2, count($entitlements));
+
+  }
+
+  public function testgetLeavePeriodEntitlementRemainder() {
+    $absence_period = $this->createAbsencePeriod(date('Y') .'-01-01', date('Y') .'-12-31');
+    $periodEntitlement = $this->createPeriodEntitlement($absence_period->id);
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+
+    // This leave request will deduct 3 days from the entitlement
+    $this->createLeaveRequestBalanceChange(
+      $periodEntitlement->type_id,
+      $periodEntitlement->contact_id,
+      $this->leaveRequestStatuses['Approved'],
+      date('YmdHis'),
+      date('YmdHis', strtotime('+2 day'))
+    );
+
+    $params = ['entitlement_id' => $periodEntitlement->id];
+    $result = $periodEntitlement::getLeavePeriodEntitlementRemainder($params);
+    $this->assertEquals(1, count($result));
+    $this->assertEquals(7, $result[0]['remainder']['current']);
+  }
+
+  public function testgetLeavePeriodEntitlementRemainderWithIncludeFuture() {
+    $absence_period = $this->createAbsencePeriod(date('Y') .'-01-01', date('Y') .'-12-31');
+    $periodEntitlement = $this->createPeriodEntitlement($absence_period->id);
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+
+    // This leave request will deduct 3 days from the entitlement
+    $this->createLeaveRequestBalanceChange(
+      $periodEntitlement->type_id,
+      $periodEntitlement->contact_id,
+      $this->leaveRequestStatuses['Waiting Approval'],
+      date('YmdHis'),
+      date('YmdHis', strtotime('+2 day'))
+    );
+
+    $params = ['entitlement_id' => $periodEntitlement->id, 'include_future' => true];
+    $result = $periodEntitlement::getLeavePeriodEntitlementRemainder($params);
+    $this->assertEquals(1, count($result));
+    $this->assertEquals(10, $result[0]['remainder']['current']);
+    $this->assertEquals(7, $result[0]['remainder']['future']);
+  }
+
+  public function testgetLeavePeriodEntitlementRemainderWithContactAndPeriodId() {
+    $absence_period = $this->createAbsencePeriod(date('Y') .'-01-01', date('Y') .'-12-31');
+    $periodEntitlement = $this->createPeriodEntitlement($absence_period->id);
+
+    $this->createLeaveBalanceChange($periodEntitlement->id, 10);
+
+    $params = ['contact_id' => $periodEntitlement->contact_id, 'period_id' => $periodEntitlement->period_id];
+    $result = LeavePeriodEntitlement::getLeavePeriodEntitlementRemainder($params);
```

Comment:
I was thinking here and it's kind of redundant to have LeavePeriodEntitlement on the method name, since that this is already the class name. It would be better to just name the method as `getRemainder()`